### ISSUE TITLE
Create backlinks from upstream PRs to dependency bump PRs

### ISSUE THREAD

**Issue by jglick** [dependabot/feedback#556]
Consider https://github.com/jenkinsci/jenkins-test-harness/pull/144 which was integrated in https://github.com/jenkinsci/plugin-pom/pull/214. The POM PR links to the library PR in its **Release notes** section. However, there is no backlink from the library PR to the POM PR. This means that someone who stumbled across the library PR (say, while debugging the issue it purported to fix) would not have an easy way to see which POM version bundled the fix; you would need to `git blame` the `pom.xml` to see when the fixed version was integrated, then `git describe --contains` to see which POM version contained that fix.

Normally this kind of thing is handled quite nicely on GitHub via backlinks. If one PR mentions another, then the mentionee gets a backlink, so you can easily navigate a complex web of changes. Unfortunately this does not currently work in Dependabot-generated PRs, I think because of the way it renders the upstream changelog in Markdown. For example, it uses

```markdown
> * Fix concurrent access to LoggerRule#getMessages ([#144](https://github-redirect.dependabot.com/jenkinsci/jenkins-test-harness/issues/144)) [@&#8203;Vlatombe](https://github.com/Vlatombe)
```

which renders as

> * Fix concurrent access to LoggerRule#getMessages ([#144](https://github-redirect.dependabot.com/jenkinsci/jenkins-test-harness/issues/144)) [@&#8203;Vlatombe](https://github.com/Vlatombe)

whereas if it simply used

```markdown
> * Fix concurrent access to LoggerRule#getMessages (https://github.com/jenkinsci/jenkins-test-harness/issues/144) @Vlatombe
```

which renders as

> * Fix concurrent access to LoggerRule#getMessages (https://github.com/jenkinsci/jenkins-test-harness/issues/144) @Vlatombe

you would not only get the backlink automatically, something like

> **dependabot-preview** referenced this pull request 1 hour ago
> Bump jenkins-test-harness from 2.51 to 2.52 – # 214 – Merged

but the _forward_ hyperlink looks nicer since it will have a hover preview.

**Comment by greysteil** [comment]
Thanks for opening this. In this case the reference removal is by design - we do a lot of extra work to remove those backlinks!

Most libraries didn't like all of the "reference spam" Dependabot PRs initially created in situations like the above. Users were even less happy - in the above example `@Vlatombe` would  get hundreds / thousands of emails from all the PRs mentioning him, and we had a number of furious users get in touch with Dependabot. Generally they assumed that Dependabot was deliberately trying to spam them.

As a result, we added a lot of sanitization to our PR bodies. We still occasionally miss a reference and have someone get in touch (sometimes abusively).

Long term, I'd like to implement a better solution for the above from within GitHub. References could be bundled up and displayed clearly in the PR timeline, for example. Short term, however, we need to keep the current sanitzation so that folks don't hunt me down and kill me.

**Comment by jglick** [comment]
I see. Could this become an option in the downstream repository’s Dependabot config? In the case I mentioned, and many others like it in our organization, there is *one* place where a `jenkinsci/jenkins-test-harness` release is normally going to get consumed—that is, `jenkinsci/plugin-pom`—so we definitely want the backlinks. Could `plugin-pom/.dependabot/config.yml` therefore say something like

```yaml
create_backlinks: true
```

?

**Comment by greysteil** [comment]
I don't think we can let the downstream repo decide this - it would have to be the upstream repo that did. I don't see why we couldn't do that, but I think a solution within GitHub (i.e., bundling up references from bots) would be a lot nicer.
